src/migrations/1668109973567-communityPolicy.ts

```diff
@@ -0,0 +1,183 @@
+    );
+
+    await queryRunner.query(
+      `ALTER TABLE \`community\` ADD \`policyId\` char(36) NOT NULL`
+    );
+
+    const communities: any[] = await queryRunner.query(
+      `SELECT id, policy FROM community`
+    );
+    for (const community of communities) {
+      const policyStr = community.policy;
+      const policy: oldCommunityPolicy = JSON.parse(policyStr);
+      const newMemberPolicy: newCommunityRolePolicy = {
+        ...policy.member,
+        parentCredentials: [],
+      };
+      const newLeadPolicy: newCommunityRolePolicy = {
+        ...policy.lead,
+        parentCredentials: [],
+      };
+      const memberStr = JSON.stringify(newMemberPolicy);
+      const leadStr = JSON.stringify(newLeadPolicy);
+      const communityPolicyID = randomUUID();
+      await queryRunner.query(
+        `INSERT INTO community_policy (id, createdDate, updatedDate, version, member, lead) VALUES ('${communityPolicyID}', NOW(), NOW(), 1, '${memberStr}', '${leadStr}')`
+      );
+      await queryRunner.query(
+        `UPDATE community SET policyId = '${communityPolicyID}' WHERE (id = '${community.id}')`
+      );
+    }
+
+    await queryRunner.query('ALTER TABLE `community` DROP COLUMN `policy`');
+
+    // Now set the parent credentials
+    const communitiesNewPolicy: any[] = await queryRunner.query(
+      `SELECT id, policyId, parentCommunityId FROM community`
+    );
+    for (const community of communitiesNewPolicy) {
+      const parentCommunityId = community.parentCommunityId;
+      if (parentCommunityId) {
+        console.log(
+          `===> Updating parent credentials for community with id: ${community.id}`
+        );
+        // Need to get the parents
+        const parentMemberCredentials: any[] = [];
+        const parentLeadCredentials: any[] = [];
+        await this.getCredentialsForCommunity(
+          parentCommunityId,
+          queryRunner,
+          parentMemberCredentials,
+          parentLeadCredentials
+        );
+        const communityPolicyResult = await queryRunner.query(
+          `SELECT id, member, lead  FROM community_policy  WHERE (id = '${community.policyId}')`
+        );
+        const communityPolicy = communityPolicyResult[0];
+
+        const memberPolicy: newCommunityRolePolicy = JSON.parse(
+          communityPolicy.member
+        );
+        memberPolicy.parentCredentials = parentMemberCredentials;
+        const memberStr = JSON.stringify(memberPolicy);
+
+        const leadPolicy: newCommunityRolePolicy = JSON.parse(
+          communityPolicy.lead
+        );
+        leadPolicy.parentCredentials = parentLeadCredentials;
+        const leadStr = JSON.stringify(leadPolicy);
+
+        await queryRunner.query(
+          `UPDATE community_policy SET member = '${memberStr}', lead = '${leadStr}' WHERE (id = '${communityPolicy.id}')`
+        );
+      }
+    }
+
+    // Todo:
```

Comment:
What about this todo?

Reply:
Ah good catch - I was struggling with gettng the FK to work on the relationship enforcment community-community-policy. Can you please have a quick look to see what I was missing?